sql/core/src/test/scala/org/apache/spark/sql/collation/IndeterminateCollationTestSuite.scala

```diff
@@ -0,0 +1,218 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.collation
+
+import org.apache.spark.{SparkRuntimeException, SparkThrowable}
+import org.apache.spark.sql.{AnalysisException, DataFrame, QueryTest, Row}
+import org.apache.spark.sql.test.SharedSparkSession
+import org.apache.spark.sql.types.StringType
+
+class IndeterminateCollationTestSuite extends QueryTest with SharedSparkSession {
+
+  val testTableName = "tst_table"
+  val dataSource = "parquet"
+
+  def withTestTable(testCode: => Unit): Unit = {
+    withTable(testTableName) {
+      sql(s"""
+           |CREATE TABLE $testTableName (
+           |  c1 STRING COLLATE UTF8_LCASE,
```

Comment:
Is there additional value in also parameterizing the two explicit collations used here and mixing them up throughout the test cases below, as opposed to always using UTF8_LCASE and UTF8_BINARY?

Reply:
I don't really see the benefit, we just want the two to be different, but when it comes to their behavior this test suite should not really care